src/util/num.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Redis Ltd. 2018 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+/**
+ * This file contains helpful functions to work with numbers, like
+ * checked addition for unsigned integers, among others.
+ * To support the "overload" for a single function using different arguments
+ * (except for the out parameter), all the arguments for addition are implicitly
+ * converted to the max-width type (uint64_t) and only then, inside the body,
+ * the checks are performed.
+*/
+
+#include "num.h"
+
+#if __has_builtin(__builtin_add_overflow)
+#define CHECKED_ADD(prefix, type) \
+bool checked_add_##prefix(const type lhs, const type rhs, type *result) { \
+return !__builtin_add_overflow(lhs, rhs, result); \
+}
+#else
+#define CHECKED_ADD(prefix, type) \
+bool checked_add_##prefix(const type lhs, const type rhs, type *result) { \
+    if (!result) { \
+        return false; \
+    } \
+\
+    const type temp_result = (type)lhs + (type)rhs; \
```

Comment:
Why do we need this "extra" cast `lhs` and `rhs` are already declared as `const type`